x-pack/plugins/enterprise_search/server/routes/enterprise_search/search_applications.ts

```diff
@@ -46,6 +48,19 @@ export function registerSearchApplicationsRoutes({ log, router }: RouteDependenc
         request.query
       )) as EnterpriseSearchApplicationsResponse;
 
+      await Promise.all(
```

Comment:
Jumping in a little late here, but we should probably add some batching or rate limiting on this `Promise.all`. We're doing one call to Elasticsearch per search application here, all of them in parallel at once.

Reply:
we are just retrieving the first page of 10 search apps - so I would say it is not necessary to batch or do some rate limiting here.

I have also checked and if there are more than 10 search apps, we are still retrieving the first page and only retrieve the next ones when the user requests them:

<img width="1914" alt="Screenshot 2023-07-31 at 18 35 27" src="https://github.com/elastic/kibana/assets/16032709/1b0546a1-e216-4f76-b484-d09524bf8ec9">